logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -171,8 +167,8 @@ private String compactJson(final String json) throws IOException {
     }
 
     // this wouldn't catch spaces in json, but that's ok for our use case here
-    private boolean isAlreadyCompacted(final String json) {
-        return PRETTY_PRINT.matchesNoneOf(json);
+    final static boolean isAlreadyCompacted(final String json) {
+        return !json.matches("(?s).*[\n\r\t].*");
```

Comment:
> I tested with basic regex and it doesn't work.

The pattern should work, the current pattern searches for `[\n\r\t]` with any characters before or after it, to get the same effect with the simplified pattern you'd have to use `find` instead of `matches`. I don't think there would be a big performance difference, but it would be clearer what's being tested here.